playwright/element-web-test.ts

```diff
@@ -195,6 +207,41 @@ export const test = base.extend<
         await bot.prepareClient(); // eagerly register the bot
         await use(bot);
     },
+
+    slidingSyncProxy: async ({ usesSlidingSyncProxy, cryptoBackend, context, config, homeserver }, use) => {
+        if (!usesSlidingSyncProxy) {
+            throw new Error("Use test.use({ usesSlidingSyncProxy: true }) before using the sliding-sync-proxy!");
+        }
+        const proxy = new SlidingSyncProxy(homeserver.config.dockerUrl);
+        const proxyInstance = await proxy.start();
+        /**
+         * When the sliding sync proxy is used, we handle the route here rather than in the page
+         * fixture because we need to add the proxy address to the config.
+         *
+         * The proper way to mutate the config for a test is by overriding the config fixture
+         * via test.use(...); that can't be done here because the sliding sync proxy is only
+         * available within beforeAll and by that point it's too late to override the fixture.
+         */
+        await context.route(`http://localhost:8080/config.json*`, async (route) => {
+            const json = {
+                ...CONFIG_JSON,
+                ...config,
+                setting_defaults: {
+                    feature_sliding_sync: true,
+                    feature_sliding_sync_proxy_url: `http://localhost:${proxyInstance.port}`,
+                },
+            };
+            if (cryptoBackend === "rust") {
+                json["features"] = {
+                    ...json["features"],
+                    feature_rust_crypto: true,
+                };
+            }
+            await route.fulfill({ json });
+        });
```

Comment:
I think this will be really easy to break, you could inject the proxy url via an init script instead?